crates/sp-domains-fraud-proof/src/fraud_proof.rs

```diff
@@ -345,8 +345,8 @@ pub enum FraudProofVariant<Number, Hash, MmrHash, DomainHeader: HeaderT> {
     InvalidTransfers(InvalidTransfersProof),
     // Dummy fraud proof only used in test and benchmark
     //
-    // NOTE: the `Dummy` must be the last variant, because the `#[cfg(..)]` will apply to
-    // all the variants after it.
+    // NOTE: the `Dummy` must be the last variant, because adding or removing it changes the
+    // encoding of all the variants after it.
```

Comment:
Neither previous nor suggested comments seem to be exactly accurate.

If order is a concern, then explicit index should be specified on all variants. Then this dummy variant will not impact encoding in any way. The fact that we need a dummy to begin with is annoying as well, but I didn't investigate if there is a nicer way about it.

Reply:
Both make sense to me. This comment was originally added because I failed to compile the test as the variant after `Dummy` was not found, but I can't reproduce it anymore.

@teor2345 Could you add `#[codec(index = ..)]` to each of the variant, please?